### ISSUE TITLE
 draw model on center of line at any angle

### ISSUE THREAD

**Issue by Gautam2010** [microsoft/maker.js#554]
How to draw model on center of line at any angle.  
It draws above the line.
<img width="58" alt="image" src="https://user-images.githubusercontent.com/29875175/215082248-8d91cd1f-13fb-4265-a0bf-204ff3ad82c5.png">

Code I am trying is as below : 
`
 
    const totalDistance = makerjs.measure.pointDistance(line.origin, line.end);
    const halfDistance = totalDistance / 2 - moduleWidth / 2;
    const angle = makerjs.angle.ofLineInDegrees(line);
    makerjs.model.rotate(module, angle);
    const _angle = makerjs.angle.ofPointInRadians(line.origin, line.end);
    const x = line.origin[0] + halfDistance * Math.cos(_angle);
    const y = line.origin[1] + halfDistance * Math.sin(_angle);

    module.origin = [x, y];
`

Expectation is I am trying to make something below. Any idea what should be the correct approch?
<img width="36" alt="image" src="https://user-images.githubusercontent.com/29875175/215084550-22b23215-a075-4372-943d-a70071512340.png">

**Comment by MontassarAbdelhedi** [comment]
This can be wrong or not the case, but I feel it's worth sharing.
When you try to draw two shapes that occupy the same space, one of them gets pushed. I suggest doing this:

`var mainModel = { models: {}, paths: {} };
mainModel[0] = line;
mainModel[1] = module;
// calculate the position x and y
module.origin = [x, y]`

**Comment by Gautam2010** [comment]
@MontassarAbdelhedi  It may works when you have no angle specified or simple case, Question is on when you have line with any angle.

**Comment by danmarshall** [comment]
First you'll want to check the origin of the model you're rotating. Some models, like [Polygon](https://maker.js.org/playground/?script=Polygon) and [Star](https://maker.js.org/playground/?script=Star) are built around an origin at the center of their paths:
<img width="204" alt="image" src="https://user-images.githubusercontent.com/11507384/215565520-a29e3459-8a48-4fff-abbd-2f57974ff227.png">
Others like [Rectangle](https://maker.js.org/playground/?script=Rectangle) and [Square](https://maker.js.org/playground/?script=Square) are built around an origin in the lower left:
<img width="230" alt="image" src="https://user-images.githubusercontent.com/11507384/215565876-5dea683a-1797-4212-8607-e08da9dcdab0.png">

If your model is symmetrical, you can convert between these modes with [zeroing and centering](https://maker.js.org/docs/intermediate-drawing/#Zeroing%20and%20Centering). If your model is not symmetric or you want an origin elswhere, you can try what @MontassarAbdelhedi suggested - "pushing" or "wrapping" your model inside another model. For example, with a Square with a side of 100:

```js
//this will have origin at lower left
var square = new makerjs.models.Square(100);
square.paths['center hole'] = new makerjs.paths.Circle([50,50], 2);

//offset the square to its center
square.origin = [-50, -50];

//rotate
makerjs.model.rotate(square, 45);

//can't set square's origin! - it needs to stay at [-50, -50] because that's where its children are referencing from.
//square.origin = [20, 0];	//bug

//so we "push" or "wrap" or "encapsulate" the rotated square into a model that we can move freely
var newModel = { 
  models: [ square ],

  //now we can move to the exact location we want
  origin: [20, 0]
};
```

**Comment by Gautam2010** [comment]
Latest version `childrenOnPath`  fixed under #549
